test/rekt/features/kafkasource/scaling.go

```diff
@@ -0,0 +1,138 @@
+
+	// Setup:
+	// Sender (eventshub) -> KafkaProxy -> KafkaTopic -> KafkaSource -> Sink (eventshub)
+	proxyName := feature.MakeRandomK8sName("proxy")
+	f.Setup("set proxy name state", func(ctx context.Context, t feature.T) {
+		state.SetOrFail(ctx, t, "proxyName", proxyName)
+	})
+
+	f.Setup("install Kafka proxy", kafkaproxy.Install(proxyName,
+		kafkaproxy.WithBootstrapServer(kafkaBootstrapUrlPlain),
+		kafkaproxy.WithTopic(topic)))
+
+	f.Setup("set topic name state", func(ctx context.Context, t feature.T) {
+		state.SetOrFail(ctx, t, "topicName", topic)
+	})
+	f.Setup("install kafka topic", kafkatopic.Install(topic))
+
+	// Setup sink
+	sinkName := feature.MakeRandomK8sName("sink")
+	f.Setup("set sink state", func(ctx context.Context, t feature.T) {
+		state.SetOrFail(ctx, t, "sinkName", sinkName)
+	})
+	f.Setup("install sink", eventshub.Install(sinkName, eventshub.StartReceiver))
+
+	// Setup source
+
+	sourceName := feature.MakeRandomK8sName("source")
+	f.Setup("set source state", func(ctx context.Context, t feature.T) {
+		state.SetOrFail(ctx, t, "sourceName", sourceName)
+	})
+
+	// options
+	ksopts := []kafkasource.CfgFn{
+		kafkasource.WithBootstrapServers([]string{kafkaBootstrapUrlPlain}),
+		kafkasource.WithTopics([]string{topic}),
+		kafkasource.WithSink(&duckv1.KReference{
+			Kind:       "Service",
+			Name:       sinkName,
+			APIVersion: "v1",
+		}, ""),
+	}
+
+	// installation
+
+	f.Setup("install a kafka source", kafkasource.Install(sourceName, ksopts...))
+
+	// Check the setup meets some requirements (not what is actually asserted)
+	f.Requirement("kafka topic must be ready", kafkatopic.IsReady(topic))
+	f.Requirement("kafka source must be ready", kafkasource.IsReady(sourceName))
+
+	f.Assert("sink receive steady stream of events, with no duplicates or missing events for 10s", sinkReceiveSteadyEvents)
+	return f
+}
+
+func sinkReceiveSteadyEvents(ctx context.Context, t feature.T) {
+	proxyName := state.GetStringOrFail(ctx, t, "proxyName")
+	sinkName := state.GetStringOrFail(ctx, t, "sinkName")
+	sourceName := state.GetStringOrFail(ctx, t, "sourceName")
+
+	// See https://github.com/knative-sandbox/eventing-kafka/issues/411
+	if test_mt_source == "1" {
+		time.Sleep(10 * time.Second)
+	}
+
+	url, err := kafkaproxy.Address(ctx, proxyName)
+	if err != nil {
+		t.Fatalf("failed to get the proxy URL: %v", err)
+	}
+
+	sender := feature.MakeRandomK8sName("sender")
+	eventshub.Install(sender,
+		eventshub.StartSenderURL(url.String()),
+		eventshub.EnableIncrementalId,
+		eventshub.InputEvent(FullEvent()),
+		eventshub.SendMultipleEvents(20, time.Second), // 20 events over 20s
+	)(ctx, t)
+
+	time.Sleep(2 * time.Second)
+
+	env := environment.FromContext(ctx)
+	source, err := kafkaclient.Get(ctx).SourcesV1beta1().KafkaSources(env.Namespace()).Get(ctx, sourceName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("failed to get the KafkaSource object: %v", err)
+	}
+	source.Spec.Consumers = pointer.Int32Ptr(130)
```

Comment:
Just a random number?

Reply:
not totally. It has to be over 100 because each pod can have 100 "virtual replicas".